File: thousands-services/IvsIdleGameShared/Repositories/Interfaces/IBoostRepository.cs
using System;
using IvsIdleGameShared.Models.Boost;

namespace IvsIdleGameShared.Repositories.Interfaces;

public interface IBoostRepository
{
    Task AddBoost(string stageId, int segment, Boost boost);
    Task AddBoosts(string stageId, int segment, Boost[] boosts);
    Task<List<BoostsSegement>> GetBoostsSegments(string stageId);
    Task<BoostsSegement> GetBoosts(string stageId, int segment);
    Task<List<BoostOutput>> GetBoosts(string stageId, int segment, string? userId);
}

File: thousands-tv/wildcard/apps/dapp/repositories/interfaces/iBoostRepository.ts
import { IBoostsSegment, IUserEventBoostSummary } from "@repo/interfaces";

export default interface IBoostRepository {
    getBoosts(stageId: string, segment: number): Promise<IBoostsSegment | null>;

    getBoostsSegments(stageId: string): Promise<IBoostsSegment[] | null>;

    createBoostsSegment(
        stageId: string,
        vendorEventId: string,
        segment: number,
        boostsCount: number,
        userPool: string[]
    ): Promise<IBoostsSegment>;

    getUserBoostSegmentSummaryByEvent(userId: string): Promise<IUserEventBoostSummary[]>;
}
